### PR TITLE
Clean fuse label when deleting dataset

### LINKED ISSUE #1057: [BUG]Clean up fuse selector label when destroying the jindoruntime
**What is your environment(Kubernetes version, Fluid version, etc.)**

**Describe the bug**

1. When JindoRuntime is deleted, the related configmap is not cleaned up.
2. In some corner cases, the node selector for fuse pod may be left in the node. For example, if the app pod is deleted in force way, the CSI plugin maybe skipped to call.

**What you expect to happen:**

**How to reproduce it**

**Additional Information**

### PR DESCRIPTION
**Pull request by cheyang**
### Ⅱ. Does this pull request fix one issue?

fixes #1057 

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: pkg/ddc/jindo/fuse.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+	labelNames := []string{fuseLabelKey}
+	e.Log.Info("check node labels", "labelNames", labelNames)
+	fuseLabelSelector, err := labels.Parse(fmt.Sprintf("%s=true", fuseLabelKey))
+	if err != nil {
+		return
+	}
+
+	err = e.List(context.TODO(), nodeList, &client.ListOptions{
+		LabelSelector: fuseLabelSelector,
```
Comment (frankleaf) [inline pkg/ddc/jindo/fuse.go:47]: will delete all nodes with fuse label? how about  other fuse is running on node with lable，will delete lable？thx

File: pkg/ddc/jindo/fuse.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	}
+
+	var labelsToModify common.LabelsToModify
+	labelsToModify.Delete(fuseLabelKey)
+
+	for _, node := range nodes {
+		_, err = utils.ChangeNodeLabelWithPatchMode(e.Client, &node, labelsToModify)
+		if err != nil {
+			e.Log.Error(err, "Error when patching labels on node", "nodeName", node.Name)
+			return count, errors.Wrapf(err, "NodeStageVolume: error when patching labels on node %s", node.Name)
```
Comment (TrafalgarZZZ) [inline pkg/ddc/jindo/fuse.go:68]:
```suggestion
			return count, errors.Wrapf(err, "error when patching labels on node %s", node.Name)
```

File: pkg/ddc/jindo/shutdown.go [hunk 3]
```diff
@@ -91,13 +92,32 @@ func (e *JindoEngine) releasePorts() (err error) {
...
+	if err != nil {
+		e.Log.Error(err, "Err in cleaning configMap")
+		return err
+	}
+	return
+}
+
+// cleanConfigmap cleans up the configmaps, such as:
+// {dataset name}-jindo-values, {dataset name}-jindofs-client-config, {dataset name}-jindofs-config
+func (e *JindoEngine) cleanConfigmap() (err error) {
```
Comment (TrafalgarZZZ) [inline pkg/ddc/jindo/shutdown.go:112]: I think it's better to use "ConfigMap"  to be consistent with existing method like `kubeclient.DeleteConfigMap`

**Review by TrafalgarZZZ — APPROVED** [review]
LGTM

File: pkg/ddc/jindo/fuse.go (hunk 1 above)
Comment (TrafalgarZZZ) [reply pkg/ddc/jindo/fuse.go:47]: The selector includes namesapce and name, which must be globally unique for the dataset and bounded jindoruntime. So I think that would not be a problem.

File: pkg/ddc/jindo/shutdown.go (hunk 3 above)
Comment (cheyang) [reply pkg/ddc/jindo/shutdown.go:112]: Thanks, I will update according to your suggestions.

**Pull request merged**